src/compiler/checker.ts

```diff
@@ -8822,6 +8959,177 @@ module ts {
             }
         }
 
+        function checkNonThenableType(type: Type, location?: Node, message?: DiagnosticMessage) {
+            if (!allConstituentTypesHaveKind(type, TypeFlags.Any) && isTypeAssignableTo(type, getGlobalThenableType())) {

```

Comment:
Per @JsonFreeman's comment, I updated `getPromisedType` to follow a similar path as `getIteratedType`, which calls `allConstituentTypesHaveKind` with `TypeFlags.Any`. As such, I updated this reference to follow the same approach.